code/__HELPERS/path.dm

```diff
@@ -0,0 +1,379 @@
+	var/atom/movable/caller
+	/// The turf where we started at
+	var/turf/start
+	/// The turf we're trying to path to (note that this won't track a moving target)
+	var/turf/end
+	/// The open list/stack we pop nodes out from
+	var/datum/heap/path/open
+	/// An assoc list that matches turfs (the key) to their nodes if said turf has one
+	var/list/openc
+	/**
+	 * An assoc list that serves as the closed list & tracks what turfs came from where. Key is the turf, and the value is what turf it came from
+	 *
+	 * Nodes are only created & added to the heap/openc once a turf is found "interesting", but due to recursion, we may not know it's interesting until we finish processing its children.
+	 * Inserting to this list is cheaper than making a node datum + inserting into the heap, so everything goes in here immediately, and what we use to recreate the path at the end
+	 */
+	var/list/sources
+	/// The list we compile at the end if successful to pass back
+	var/list/path
+
+	// general pathfinding vars/args
+	/// An ID card representing what access we have and what doors we can open. Its location relative to the pathing atom is irrelevant
+	var/obj/item/card/id/id
+	/// How far away we have to get to the end target before we can call it quits
+	var/mintargetdist = 0
+	/// I don't know what this does vs maxnodes, but they limit how far we can search before giving up on a path
+	var/maxnodedepth = 30
+	/// I don't know what this does vs maxnodedepth, but they limit how far we can search before giving up on a path
+	var/maxnodes = 30
+	/// The proc we use to tell the distance between two turfs. Currently ignored in favor of get_dist/get_dist
+	///var/dist = /turf/proc/Distance
+	/// Do we only worry about turfs with simulated atmos, most notably things that aren't space?
+	var/simulated_only
+	/// Did we succeed?
+	var/success = FALSE
+	/// A specific turf we're avoiding
+	var/turf/avoid
+
+/datum/pathfind/New(atom/movable/caller, atom/goal, id, maxnodes, maxnodedepth, mintargetdist, simulated_only, avoid)
+	src.caller = caller
+	end = get_turf(goal)
+	open = new()
+	openc = new() //open list for node check
+	sources = new() //open list for node check
+	src.id = id
+	src.maxnodes = maxnodes
+	src.maxnodedepth = maxnodedepth
+	src.mintargetdist = mintargetdist
+	src.simulated_only = simulated_only
+	src.avoid = avoid
+
+/// The proc you use to start the search, returns FALSE if it's invalid, an empty list if no path could be found, or a valid path to the target
+/datum/pathfind/proc/start_search()
+	caller.calculating_path = TRUE
+	start = get_turf(caller)
+	if(!start || !end)
+		stack_trace("Invalid A* start or destination")
+		return FALSE
+	if(start.z != end.z || start == end ) //no pathfinding between z levels
+		return FALSE
+	if(maxnodes) //if start turf is farther than maxnodes from end turf, no need to do anything
+		maxnodedepth = maxnodes
+
+	//initialization
+	var/datum/jps_node/cur = new (start,null,0,end)//current processed turf
```

Comment:
```suggestion
	var/datum/jps_node/current_processed_turf = new (start, null, 0, end)
```
Or `current_turf`
No need to go cheap on spaces nor on names.
Readability over writability (and if you are using VSC you get auto-complete anyway).